view/frontend/web/insights.js

```diff
@@ -0,0 +1,250 @@
+requirejs([
+    'jquery',
+    'algoliaAnalytics',
+    'algoliaBundle',
+], function ($, algoliaAnalytics, algoliaBundle) {
+
+    var algoliaInsights = {
+        config: null,
+        defaultIndexName: null,
+
+        track: function(algoliaConfig) {
+
+            this.config = algoliaConfig;
+            this.defaultIndexName = algoliaConfig.indexName + '_products';
+
+            if (algoliaConfig.ccAnalytics.enabled
+                || algoliaConfig.personalization.enabled) {
+
+                this.initializeAnalytics();
+                this.addSearchParameters();
+                this.bindData();
+                this.bindEvents();
+            }
+        },
+
+        initializeAnalytics: function() {
+            algoliaAnalytics.init({
+                appId: this.config.applicationId,
+                apiKey: this.config.apiKey
+            });
+
+            var userAgent = 'insights-js-in-magento (' + this.config.extensionVersion + ')';
+            algoliaAnalytics.addAlgoliaAgent(userAgent);
+
+            var userToken = getCookie('aa-search');
+            if (userToken && userToken != '') algoliaAnalytics.setUserToken(userToken);
```

Comment:
it's best to always use triple equal sign operators: `userToken !== ''` instead of `userToken != ''`. 

loose equality/inequality are often hard to reason on. 

```
false != 0   // -> false 
false !== 0 // -> true
```

Reply:
Aha this is the same best practice for PHP. Will update this!